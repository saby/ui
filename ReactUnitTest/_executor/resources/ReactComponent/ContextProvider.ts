import testContext from './testContext';

export default testContext.Provider;
