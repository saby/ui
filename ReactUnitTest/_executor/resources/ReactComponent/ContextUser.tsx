import { useContext } from 'react';
import testContext from './testContext';

export default function ContextUser(): JSX.Element {
    const testContextValue = useContext(testContext);
    return <div className="contextUser">Context value: {testContextValue}</div>;
}
