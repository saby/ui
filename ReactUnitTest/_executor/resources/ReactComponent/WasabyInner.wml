<div>Is Wasaby Control</div>
