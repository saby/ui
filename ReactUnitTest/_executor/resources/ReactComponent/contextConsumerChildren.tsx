export default function contextConsumerChildren(value: string) {
    return <div className="contextConsumerChildren">Context value: {value}</div>;
}
