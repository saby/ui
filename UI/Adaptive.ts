/**
 * Библиотека для настройки адаптивности сайта к различным аспектам адаптивности
 * @see https://n.sbis.ru/wasaby/knowledge#toc_2eb25e0a-c867-45a8-a13d-b8c1ff12cdbc
 * @library UICore/Adaptive
 * @public публичная библиотека
 */

export {
    DEFAULT_BREAKPOINTS,
    AdaptiveContainer,
    AdaptiveInitializerConfig,
    AdaptiveModeType,
    useAdaptiveMode,
    withAdaptiveMode,
    getAdaptiveModeForLoaders,
    unsafe_getRootAdaptiveMode,
    ScrollOnBodyStore,
    BodyScroll,
    moveBodyScroll,
    unMoveBodyScroll,
    ScrollOnBodyContext,
    ScrollOnBodyContextProvider,
} from 'UICore/Adaptive';
