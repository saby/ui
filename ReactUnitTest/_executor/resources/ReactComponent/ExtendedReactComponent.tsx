import ReactComponent from './ReactComponent';

// Реализация render наследуется от ReactComponent.
export default class ExtendedReactComponent extends ReactComponent {
    protected text: string = 'Is Extended React Component';
}
