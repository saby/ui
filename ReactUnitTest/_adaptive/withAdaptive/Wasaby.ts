import { Control, TemplateFunction } from 'UI/Base';
import { withAdaptiveMode } from 'UICore/Adaptive';
import * as template from 'wml!ReactUnitTest/_adaptive/withAdaptive/Wasaby';

class Wasaby extends Control {
    _template: TemplateFunction = template;
}

export default withAdaptiveMode(Wasaby);
