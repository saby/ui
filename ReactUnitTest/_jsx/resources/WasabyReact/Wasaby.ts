import { Control, TemplateFunction } from 'UI/Base';
/* eslint-disable no-console */
// @ts-ignore
import * as template from 'wml!ReactUnitTest/_jsx/resources/WasabyReact/Wasaby';

export default class Wasaby extends Control {
    protected _template: TemplateFunction = template;

    handler() {
        this._notify('valueChanged', [
            'bind handler ' + this._options.additionName,
        ]);
    }
}
