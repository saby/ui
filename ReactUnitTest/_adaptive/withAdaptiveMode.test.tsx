/**
 * @jest-environment jsdom
 */
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import { AdaptiveContainer, Storage } from 'UICore/Adaptive';

import ReactComponent from './withAdaptive/ReactComponent';
import Wasaby from './withAdaptive/Wasaby';

const strangeSizeToTest = 31415;

describe('withAdaptive HOC', () => {
    let container: HTMLDivElement;
    beforeEach(() => {
        container = document.createElement('div');
        document.body.appendChild(container);
    });
    afterEach(() => {
        Storage.getInstance()._clear();
        unmountComponentAtNode(container);
        container.remove();
        container = null;
    });

    test('Wasaby control', () => {
        act(() => {
            render(
                <AdaptiveContainer width={strangeSizeToTest}>
                    <Wasaby />
                </AdaptiveContainer>,
                container
            );
        });
        expect(container).toMatchSnapshot();
    });
    test('Class react component', () => {
        act(() => {
            render(
                <AdaptiveContainer width={strangeSizeToTest}>
                    <ReactComponent />
                </AdaptiveContainer>,
                container
            );
        });
        expect(container).toMatchSnapshot();
    });
});
