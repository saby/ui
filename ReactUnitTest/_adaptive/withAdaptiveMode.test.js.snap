// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withAdaptive HOC Class react component 1`] = `
<div>
  <div
    class="reactWithAdaptiveMode"
  >
    <div>
      31415
    </div>
  </div>
</div>
`;

exports[`withAdaptive HOC Wasaby control 1`] = `
<div>
  <div
    class="wasabyWithAdaptiveMode"
    tabindex="0"
  >
    <div>
      31415
    </div>
  </div>
</div>
`;
